### PR TITLE
ZNTA-2224: tidy editor css

### PR DESCRIPTION
**Pull request by kathryngo**
JIRA issue URL: https://zanata.atlassian.net/browse/ZNTA-2224

The current suitcss naming convention cannot be used in editor as it will conflict with the frontend css. This can be resolved in the future by combining both frontend and editor css and prefixing all editor css with 'editor'. 

To keep a level of consistency throughout editor css, the preexisting naming convention has been used, *unless* the css is for bootstrap components uses frontend component (suitcss) styling.

## Checklist

- JIRA link
- Check target branch
- Make sure all commit statuses are green or otherwise documented reasons to ignore
- QA needs to evaluate against the JIRA ticket
- Changed files and commits make sense for this PR

See [Zanata Development Guidelines](https://github.com/zanata/zanata-platform/wiki/Development-Guidelines) more for information.

----
*This template can be updated in .github/PULL_REQUEST_TEMPLATE.md*

---
This change is [Reviewable](https://reviewable.io/reviews/zanata/zanata-platform/578)

### CODE REVIEW

File: server/zanata-frontend/src/frontend/app/styles/style.less [hunk 1]
```diff
@@ -2799,7 +2799,6 @@ button.btn.btn-danger.dropdown-toggle>span.caret, button.btn.btn-info.dropdown-t
 // Allow for dropdowns to go bottom up (aka, dropup-menu)
 //
 // Just add .dropup after the standard .dropdown class and you're set, bro.
-// TODO: abstract this so that the navbar fixed styles are not placed here?
```
Comment (kathryngo) [inline server/zanata-frontend/src/frontend/app/styles/style.less]: since this is default bootstrap css it should stay in this file

**Comment by davidmason** [comment]
If it is just specific component name clashes that are preventing using a consistent naming scheme, it might be better to prefix all the components that are duplicated in the editor with `Editor` then use consistent naming. +(Under review)  -(Awaiting review) 

---

Reviewed 29 of 31 files at r1, 6 of 6 files at r2.
Review status: all files reviewed at latest revision, 5 unresolved discussions.

---

*[server/zanata-frontend/src/frontend/app/editor/components/Button/index.css, line 30 at r2](https://reviewable.io:443/reviews/zanata/zanata-platform/578#-KwIxPNu0TnAvj-2VmCo:-KwIxPNu0TnAvj-2VmCp:b-4uvmpy) ([raw file](https://github.com/zanata/zanata-platform/blob/356b89d51dc7269f93dff568093f805e0c9c46c3/server/zanata-frontend/src/frontend/app/editor/components/Button/index.css#L30)):*
> ```CSS
> /* Need to stick to this naming convention (against suitcss) so that it does
> not conflict with the other frontend code
>  */
> ```

Is that just for buttons? If so we could use `EditorButton`, which makes it clearer which is which (with a TODO to combine them when we can).

---

*[server/zanata-frontend/src/frontend/app/editor/components/Dropdown/index.css, line 22 at r2](https://reviewable.io:443/reviews/zanata/zanata-platform/578#-KwIyVZCVHHYBRReGGSB:-KwIyVZCVHHYBRReGGSC:b-rfzwud) ([raw file](https://github.com/zanata/zanata-platform/blob/356b89d51dc7269f93dff568093f805e0c9c46c3/server/zanata-frontend/src/frontend/app/editor/components/Dropdown/index.css#L22)):*
> ```CSS
> /* Need to stick to this naming convention (against suitcss) so that it does
> not conflict with the other frontend code
>  */
> ```

Could we call it `EditorDropdown` to distinguish it and allow consistent naming?

---

*[server/zanata-frontend/src/frontend/app/editor/components/FilterToggle/index.css, line 13 at r2](https://reviewable.io:443/reviews/zanata/zanata-platform/578#-KwIyqcrMrjHlJ3q63xQ:-KwIyqcrMrjHlJ3q63xR:beku5zs) ([raw file](https://github.com/zanata/zanata-platform/blob/356b89d51dc7269f93dff568093f805e0c9c46c3/server/zanata-frontend/src/frontend/app/editor/components/FilterToggle/index.css#L13)):*
> ```CSS
> 
> /* Need to stick to this naming convention (against suitcss) so that it does
> not conflict with the other frontend code
> ```

This should be `FilterToggle` anyway, to be consistent with the component name. Is that enough to distinguish it?

---

*[server/zanata-frontend/src/frontend/app/editor/components/InputGroup/index.css, line 11 at r2](https://reviewable.io:443/reviews/zanata/zanata-platform/578#-KwIz5lnEGruqbRgOhnf:-KwIz5lnEGruqbRgOhng:b87ql5k) ([raw file](https://github.com/zanata/zanata-platform/blob/356b89d51dc7269f93dff568093f805e0c9c46c3/server/zanata-frontend/src/frontend/app/editor/components/InputGroup/index.css#L11)):*
> ```CSS
> 
> /* Need to stick to this naming convention (against suitcss) so that it does
> not conflict with the other frontend code
> ```

Could we use `EditorInputGroup` instead?

---

*[server/zanata-frontend/src/frontend/app/editor/containers/Root/index.css, line 886 at r2](https://reviewable.io:443/reviews/zanata/zanata-platform/578#-KwIzqSOrMdx9XzuGg0W:-KwIzqSOrMdx9XzuGg0X:bffwxiu) ([raw file](https://github.com/zanata/zanata-platform/blob/356b89d51dc7269f93dff568093f805e0c9c46c3/server/zanata-frontend/src/frontend/app/editor/containers/Root/index.css#L886)):*
> ```CSS
> 
> /* This needs to be this specific to target a certain element in the editor that
> does not have a custom class */
> ```

Could we just add a custom class on it instead? This is really super-fragile - change the colour and the vertical alignment gets broken.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-platform/578#-:-KwJ07jbcDXXlGA4PUAX:b-tlztfv)*

**Comment by kathryngo** [comment]
Some clash, some don't. For the purpose of consistency I am making them all use capital letters, so that way there is absolutely no conflict with frontend.  I will prefix all the editor css with editor when I do the merge with frontend css.

---

Review status: all files reviewed at latest revision, 5 unresolved discussions.

---

*[server/zanata-frontend/src/frontend/app/editor/components/Button/index.css, line 30 at r2](https://reviewable.io:443/reviews/zanata/zanata-platform/578#-KwIxPNu0TnAvj-2VmCo:-KwJ1JunKQEvJKL8IyTg:b-80tlqk) ([raw file](https://github.com/zanata/zanata-platform/blob/356b89d51dc7269f93dff568093f805e0c9c46c3/server/zanata-frontend/src/frontend/app/editor/components/Button/index.css#L30)):*
<details><summary><i>Previously, davidmason (David Mason) wrote…</i></summary><blockquote>

Is that just for buttons? If so we could use `EditorButton`, which makes it clearer which is which (with a TODO to combine them when we can).
</blockquote></details>

Sure.

---

*[server/zanata-frontend/src/frontend/app/editor/components/Dropdown/index.css, line 22 at r2](https://reviewable.io:443/reviews/zanata/zanata-platform/578#-KwIyVZCVHHYBRReGGSB:-KwJ1LYcyrSwAbrKznmq:b-80tlqk) ([raw file](https://github.com/zanata/zanata-platform/blob/356b89d51dc7269f93dff568093f805e0c9c46c3/server/zanata-frontend/src/frontend/app/editor/components/Dropdown/index.css#L22)):*
<details><summary><i>Previously, davidmason (David Mason) wrote…</i></summary><blockquote>

Could we call it `EditorDropdown` to distinguish it and allow consistent naming?
</blockquote></details>

Sure.

---

*[server/zanata-frontend/src/frontend/app/editor/components/FilterToggle/index.css, line 13 at r2](https://reviewable.io:443/reviews/zanata/zanata-platform/578#-KwIyqcrMrjHlJ3q63xQ:-KwJ1MVVMCryMY9QhkXw:b2nign3) ([raw file](https://github.com/zanata/zanata-platform/blob/356b89d51dc7269f93dff568093f805e0c9c46c3/server/zanata-frontend/src/frontend/app/editor/components/FilterToggle/index.css#L13)):*
<details><summary><i>Previously, davidmason (David Mason) wrote…</i></summary><blockquote>

This should be `FilterToggle` anyway, to be consistent with the component name. Is that enough to distinguish it?
</blockquote></details>

yep

---

*[server/zanata-frontend/src/frontend/app/editor/components/InputGroup/index.css, line 11 at r2](https://reviewable.io:443/reviews/zanata/zanata-platform/578#-KwIz5lnEGruqbRgOhnf:-KwJ1Sozg4ir73_8EMJ4:bdxvwym) ([raw file](https://github.com/zanata/zanata-platform/blob/356b89d51dc7269f93dff568093f805e0c9c46c3/server/zanata-frontend/src/frontend/app/editor/components/InputGroup/index.css#L11)):*
<details><summary><i>Previously, davidmason (David Mason) wrote…</i></summary><blockquote>

Could we use `EditorInputGroup` instead?
</blockquote></details>

Sure. Will prefix everything else during the merge.

---

*[server/zanata-frontend/src/frontend/app/editor/containers/Root/index.css, line 886 at r2](https://reviewable.io:443/reviews/zanata/zanata-platform/578#-KwIzqSOrMdx9XzuGg0W:-KwJ1QUk7v3-xbD4B7vC:bxl5uqe) ([raw file](https://github.com/zanata/zanata-platform/blob/356b89d51dc7269f93dff568093f805e0c9c46c3/server/zanata-frontend/src/frontend/app/editor/containers/Root/index.css#L886)):*
<details><summary><i>Previously, davidmason (David Mason) wrote…</i></summary><blockquote>

Could we just add a custom class on it instead? This is really super-fragile - change the colour and the vertical alignment gets broken.
</blockquote></details>

Will give it a go

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-platform/578#-:-KwJ12bvEGLrOrq9SKp0:b-48hgdt)*

**Comment by davidmason** [comment]
:lgtm: -(Under review)  +Reviewed 

---

Reviewed 31 of 31 files at r3.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-platform/578#-:-KwXn1F28iPmqgZKG9-K:b-uf1ptv)*

**Comment by spathare21** [comment]
Verified editor is working fine

**Pull request merged**